homeassistant/components/hue/light.py

```diff
@@ -380,7 +382,7 @@ def device_info(self):
         elif effect == EFFECT_RANDOM:
             command['hue'] = random.randrange(0, 65535)
             command['sat'] = random.randrange(150, 254)
-        elif self.is_philips:
+        elif self.is_philips or effect == '':
```

Comment:
What about `elif self.is_philips or not effect:`

Reply:
I don't think that works. So when you switch on the light the effect is `None`. This will break the stuff again like mentioned in @awarecan's post for older Osram bulbs.

If you don't want to define the "empty" effect then this might work:
`elif self.is_philips or effect:`